scripts/gen-cddl.sh

```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+changed=$(git diff --name-only origin/master -- 'eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/CDDL.hs')
+
+if [[ -n "$changed" ]]; then
+  cabal run cardano-cddl
```

Comment:
`gen` prefix could be a bit ambiguous and overused, so maybe alternative names for the executable could be `cuddle-cddl` or `huddle-cddl`.
```suggestion
  cabal run cardano-ledger-conway:gen-cddl
```